modules/azurerm_vm/main.tf

```diff
@@ -83,11 +83,20 @@ resource "azurerm_linux_virtual_machine" "vm" {
 
   source_image_reference {
     publisher = var.os_publisher
-    offer     = var.os_offer
-    sku       = var.os_sku
+    offer     = var.fips_enabled ? "0001-com-ubuntu-pro-focal-fips" : var.os_offer
+    sku       = var.fips_enabled ? "pro-fips-20_04-gen2" : var.os_sku
     version   = var.os_version
   }
 
+  dynamic "plan" {
+    for_each = var.fips_enabled ? [1] : []
+    content {
+      name       = "pro-fips-20_04-gen2"
+      publisher  = "canonical"
+      product    = "0001-com-ubuntu-pro-focal-fips"
```

Comment:
Same comment. This might be in the SAS subscription but others may have to pay for this. Just need to verify.

Reply:
Same as above.